src/libraries/System.IO.FileSystem/tests/FileStream/ctor_str_fm.cs

```diff
@@ -213,11 +214,23 @@ public void FileModeTruncateExisting(string streamSpecifier)
         [Theory, MemberData(nameof(StreamSpecifiers))]
         public virtual void FileModeAppend(string streamSpecifier)
         {
-            using (FileStream fs = CreateFileStream(GetTestFilePath() + streamSpecifier, FileMode.Append))
+            string fileName = GetTestFilePath() + streamSpecifier;
+            using (FileStream fs = CreateFileStream(fileName, FileMode.Append))
             {
                 Assert.False(fs.CanRead);
                 Assert.True(fs.CanWrite);
+
+                fs.Write(Encoding.ASCII.GetBytes("abcde"));
+                Assert.Equal(5, fs.Length);
+                Assert.Equal(5, fs.Position);
+                Assert.Equal(1, fs.Seek(1, SeekOrigin.Begin));
```

Comment:
I would expect that `FileMode.Append` means "I want to append data to EOF" rather than "I want to write data at requested offset". The official [doc](https://docs.microsoft.com/en-us/dotnet/api/system.io.filemode?view=net-5.0) for `FileMode.Append` says:

> Trying to seek to a position before the end of the file throws an IOException exception

I believe that current implementation of `Append` is buggy and we should fix it instead of adding a test that ensures that it's buggy. I am going to reopen https://github.com/dotnet/runtime/pull/55465 for that

Reply:
I interpret that to mean before the end of the file at the time it was opened, and the current implementation is consistent with that and has been for 20 years.

Reply:
> adding a test that ensures that it's buggy

The tests validate the expected behavior.